background.js

```diff
@@ -1,49 +1,82 @@
-browser.pageAction.onClicked.addListener(async () => {
-  await browser.tabs.executeScript({
-    code: `document.getElementsByClassName("download")[0].click();`,
-  });
+const showMessage = async message => {
+  await browser.tabs.executeScript({code: `
+    closeBtn = document.createElement("span");
+    closeBtn.textContent = "×";
+    closeBtn.setAttribute("style", "font-weight:bold;float:right;font-size:20px;line-height:18px;cursor:pointer;padding-left:15px;");
+    closeBtn.setAttribute("onclick", "this.parentElement.style.display='none';");
+    alertBox = document.createElement("div");
+    alertBox.setAttribute("style", "padding: 20px;background-color:#f44336;color:white;margin-bottom:15px;transition: opacity 0.3s linear 2s;opacity:0.83;");
+    alertBox.textContent = "${message}";
+    alertBox.appendChild(closeBtn);
+    document.querySelector(".caal-alerts").appendChild(alertBox);
+  `});
+}
+
+const clickElement = async querySelector => {
+  await browser.tabs.executeScript({code: `
+    if (document.querySelector("${querySelector}"))
+      document.querySelector("${querySelector}").click();
+  `});
+}
+
+const waitForElement = async querySelector => {
+  const elementFoundScript = `document.querySelector("${querySelector}") !== null;`;
+  const retryDelay = 100;
+
+  let timeLeft = 1000;
+  let [isElement] = await browser.tabs.executeScript({code: elementFoundScript});
+
+  while (!isElement && timeLeft > 0) {
+    await new Promise(r => setTimeout(r, retryDelay));
+    [isElement] = await browser.tabs.executeScript({code: elementFoundScript});
+    timeLeft -= retryDelay;
+  }
 
-  await new Promise(r => setTimeout(r, 500));
+  if (!isElement)
+    showMessage(`Cannot find \`${querySelector}\` element!`);
```

Comment:
Minor:

If you return a rejected promise if there is an unrecoverable failure (e.g. an element cannot be found), this will stop the promise chain and next operations won't be executed. This is a way to stop executing the rest of the code if there was a failure. See <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#error_propagation> for more information